Index: core/src/main/java/org/apache/oozie/coord/CoordELFunctions.java
===================================================================
--- core/src/main/java/org/apache/oozie/coord/CoordELFunctions.java	(revision 1506445)
+++ core/src/main/java/org/apache/oozie/coord/CoordELFunctions.java	(working copy)
@@ -40,8 +40,8 @@

 public class CoordELFunctions {
     final private static XLog LOG = XLog.getLog(CoordELFunctions.class);
-    final private static String DATASET = "oozie.coord.el.dataset.bean";
-    final private static String COORD_ACTION = "oozie.coord.el.app.bean";
+    final public static String DATASET = "oozie.coord.el.dataset.bean";
+    final public static String COORD_ACTION = "oozie.coord.el.app.bean";
     final public static String CONFIGURATION = "oozie.coord.el.conf";
     final public static String LATEST_EL_USE_CURRENT_TIME = "oozie.service.ELService.latest-el.use-current-time";
     // INSTANCE_SEPARATOR is used to separate multiple directories into one tag.
